src/app/services/data.service.ts

```diff
@@ -147,6 +150,35 @@ export class DataService {
     this.setStorage();
   }
 
+  setLastRoute(route: Route): void {
+    this.data.lastRoute = JSON.parse(JSON.stringify(route));
+    this.setStorage();
+  }
+
+  getLastRoute(): Route {
+    return this.data.lastRoute;
+  }
+
+  resetLastRoute(): void {
+    delete this.data.lastRoute;
+    this.setStorage();
```

Comment:
in line 88 you set the default lastRoute to null and here your remove the attribute completely.

could it be set to null as well here?